libs/langgraph/langgraph/prebuilt/tool_node.py

```diff
@@ -328,12 +383,20 @@ def __init__(self, field: Optional[str] = None) -> None:
 def _get_state_args(tool: BaseTool) -> Dict[str, Optional[str]]:
     full_schema = tool.get_input_schema()
     tool_args_to_state_fields: Dict = {}
+
+    def _is_injection(type_arg: Any):
```

Comment:
Was failing to infer for generics (`Annotated[T, InjectedState()]`) since that is resolved as `Optional[ANnotated....]`)